src/manifolds/HeisenbergMatrices.jl

```diff
@@ -0,0 +1,140 @@
+@doc raw"""
+    HeisenbergMatrices{T} <: AbstractDecoratorManifold{𝔽}
+
+Heisenberg matrices `HeisenbergMatrices(n)` is the manifold of ``(n+2)×(n+2)`` matrices [BinzPods:2008](@cite)
+
+```math
+\begin{bmatrix} 1 & \mathbf{a} & c \\
+\mathbf{0} & I_n & \mathbf{b} \\
+0 & \mathbf{0} & 1 \end{bmatrix}
+```
+
+where ``I_n`` is the ``n×n`` unit matrix, ``\mathbf{a}`` is a row vector of length ``n``,
+``\mathbf{b}`` is a column vector of length ``n`` and ``c`` is a real number.
```

Comment:
````suggestion
```math
\begin{bmatrix} 1 & \mathbf{a} & c \\
\mathbf{0}_N & I_n & \mathbf{b} \\
0 & \mathbf{0}_n & 1 \end{bmatrix}
```

where ``I_n`` is the ``n×n`` unit matrix, ``\mathbf{a}`` is a row vector of length ``n``,
``\mathbf{b}`` is a column vector of length ``n``, ``\mathbf{0}_n`` is the zero-vector of length ``n``,  and ``c`` is a real number.
```
Maybe this is consistent with `I_n` ? It could also be shortly mentioned int the `where`-lines

Reply:
I don't think your edit is entirely right. There is one column 0 vector and one row 0 vector. Would you prefer to use transposition to differentiate between them? Or maybe give both dimensions in subscript?

Reply:
ah and I have a N/n problem. I would prefer it to be a column vector I think, then – yes – the line 9 one should get a transpose.

Reply:
I think I've applied it correctly.